### PR TITLE
Makes CTF flags and spawn protection not affect or get affected by non CTF mobs.

### PR DESCRIPTION
**Pull request by silicons**
Not really a feature :P 
Also adds stunproofing to CTF mobs.

### CODE REVIEW

File: code/modules/awaymissions/capture_the_flag.dm [hunk 1]
```diff
@@ -534,6 +534,8 @@
 	return
 
 /obj/structure/trap/ctf/trap_effect(mob/living/L)
+	if(!is_ctf_target(L))
+		return
```
Comment (coiax) [inline code/modules/awaymissions/capture_the_flag.dm]: This is already covered by the faction check.

File: code/modules/awaymissions/capture_the_flag.dm [hunk 2]
```diff
@@ -48,7 +48,7 @@
 		STOP_PROCESSING(SSobj, src)
 
 /obj/item/weapon/twohanded/ctf/attack_hand(mob/living/user)
-	if(!user)
+	if(!is_ctf_target(user))
 		return
```
Comment (coiax) [inline code/modules/awaymissions/capture_the_flag.dm]: Also, should probably give them a message.

**Comment by KorPhaeron** [comment]
What are non ctf mobs doing there?

**Comment by kevinz000** [comment]
@KorPhaeron coiax ideas of station ctf that I tried out during a semi meme round.

File: code/modules/awaymissions/capture_the_flag.dm (hunk 1 above)
Comment (kevinz000) [inline code/modules/awaymissions/capture_the_flag.dm]: Are we sure? Because it kills anyone WITHOUT the CTF faction, don't all mobs lack it?

Comment (kevinz000) [inline code/modules/awaymissions/capture_the_flag.dm]: yeah I just tested it doesn't cover it.

File: code/modules/awaymissions/capture_the_flag.dm [hunk 3]
```diff
@@ -48,7 +48,8 @@
 		STOP_PROCESSING(SSobj, src)
 
 /obj/item/weapon/twohanded/ctf/attack_hand(mob/living/user)
-	if(!user)
```
Comment (optimumtact) [inline code/modules/awaymissions/capture_the_flag.dm]: You need to check that it's actually a mob/living with if(istype(user)) just specifying the type path in the variable argument is never enough.

Does is_ctf_target do that?

Comment (kevinz000) [inline code/modules/awaymissions/capture_the_flag.dm]: yep

**Pull request merged**